docs/06/01/types.md

```diff
@@ -0,0 +1,159 @@
+  currency?: string
+  theme?: DeepPartial<GTheme>
+  queryClient?: QueryClient
+  components: ComponentsContextType
+  children?: React.ReactNode
+}
+```
+
+The key props are:
+
+- `config`: Configuration for the API connection
+- `components`: Your custom component adapter implementing `ComponentsContextType`
+- `theme`: (Optional) Custom theming options
+- `dictionary`: (Optional) Translation overrides
+- `lng`: (Optional) Language setting
+- `locale` and `currency`: (Optional) Localization settings
+
+### Component Prop Types
+
+Each UI component has its own prop interface. Here are some of the most commonly used ones:
+
+#### TextInputProps
+
+From [TextInputTypes.ts](https://github.com/Gusto/embedded-react-sdk/blob/main/src/components/Common/UI/TextInput/TextInputTypes.ts):
+
+```typescript
+export interface TextInputProps {
+  label: string
+  description?: ReactNode
+  errorMessage?: ReactNode
+  isRequired?: boolean
+  isDisabled?: boolean
+  isInvalid?: boolean
+  id?: string
+  name: string
+  value?: string
+  placeholder?: string
+  onChange?: (value: string) => void
+  onBlur?: () => void
+  inputRef?: React.RefObject<HTMLInputElement>
+  shouldVisuallyHideLabel?: boolean
+  // Plus any additional HTML input props
+}
+```
+
+#### ButtonProps
+
+From [ButtonTypes.ts](https://github.com/Gusto/embedded-react-sdk/blob/main/src/components/Common/UI/Button/ButtonTypes.ts):
+
+```typescript
+export interface ButtonProps {
+  children: ReactNode
+  isDisabled?: boolean
+  isLoading?: boolean
+  isError?: boolean
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
+  type?: 'button' | 'submit' | 'reset'
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'link'
+  size?: 'small' | 'medium' | 'large'
+  className?: string
+  // Plus any additional HTML button props
+}
+```
+
+### Importing Types
+
+You can import these types from the SDK package:
+
+```typescript
+import type {
+  ComponentsContextType,
+  ButtonProps,
+  TextInputProps,
+  // ... other types as needed
+} from '@gusto/embedded-react-sdk'
+```
+
+The SDK exports all component types, so you don't need to access the internal file structure to use them in your application.
+
+### Type Safety
+
+The Component Adapter system leverages TypeScript to ensure type safety:
+
+1. **Compile-time checking**: TypeScript will flag any missing or incorrect props in your component implementations
+2. **IDE support**: You get autocomplete and documentation for all required props
+3. **Type inference**: TypeScript can infer the types of your event handlers and other callback functions
+
+By following the type definitions, you can ensure that your custom components will integrate seamlessly with the SDK's business logic.
+
+### Type Hierarchy
+
+The component types often extend basic HTML element props to provide both the SDK-specific props and standard HTML attributes:
+
+```typescript
+// Simplified example
+export interface TextInputProps extends HTMLInputAttributes {
```

Comment:
I think we actually got away from this in favor of only picking specific props? we didn't want there to be a lot of noise and uncertainty around what props the consumer needed to implement so we felt it might be better to just pick the ones we needed. If there are interfaces that extend the HTML attributes we might consider updating those